Fix functionality and accessibility for expand/collapse behaviour in UI Filter

tfamula opened the pull request:
Hi UI Coordinators!

in ILIAS 10, the expand/collapse behaviour in Filters does not work anymore due to the absence of Bootstrap (see [42827)](https://mantis.ilias.de/view.php?id=42827).
I decided to not adopt the old Bootstrap behaviour, but to refactor the expand/collapse behaviour, based on https://github.com/ILIAS-eLearning/ILIAS/pull/7332. At the same time, this is also a big improvement regarding accessibility.
Please have a look. If accepted, this must be picked to trunk, too. One can consider if this could also be picked to release 9 and 8 (because of the A11y improvements), but I leave that up to you.

@tfamula 

PS: The unit tests are failing because I have not adjusted the rendering tests yet. I would do that as a last step.

tfamula committed:
Fix functionality and accessibility for expand/collapse behaviour in UI Filter

thibsy reviewed (commented):
Hi @tfamula,

Thx for your contribution to the UI framework and getting rid of some bootstrap remnants!

About the concrete changes, please answer:

- [x] `onAjaxCmd`: can you explain why you remove this method from the filter's client-side public interface? Are there no usages anymore?

Kind regards,
@thibsy

tfamula commented:
Hi @thibsy,
correct! This has been used formerly for the Bulky Buttons ([expand button](https://github.com/ILIAS-eLearning/ILIAS/pull/8555/files#diff-b587a5279f811b4bca3f4632fd807154551130cc98813ce024d201e44ff44132L105) and [collapse button](https://github.com/ILIAS-eLearning/ILIAS/pull/8555/files#diff-b587a5279f811b4bca3f4632fd807154551130cc98813ce024d201e44ff44132L116)). Now, the Bulky Buttons are not needed anymore, because I implemented the `<button>` element directly into the template and visibility of elements is handled internally now in the JS file.
The Ajax requests are still needed to save the current expand/collapse state. That's why there is still the method [`performAjaxCmd`](https://github.com/ILIAS-eLearning/ILIAS/pull/8555/files#diff-ebb3bc21a4bb713ec55d79cdb24b6f839b821711beaf9b9a8b0ebffebe947152R348). It works quite similar to the old method, but I renamed it to make clear it is no part of the filter's client-side public interface anymore.

thibsy commented:
@tfamula thx for the explanation. This LGTM then. Please go ahead and amend the unit tests, so we can merge this :).

tfamula committed:
Fix rendering tests

tfamula committed:
Fix copyright

tfamula commented:
@thibsy Ready to merge!
